src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -24,7 +24,12 @@ public LinkedInAuthenticationHandler([NotNull] HttpClient client)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
-            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            var endpoint = string.Empty;
+            if (Options.Fields.Count > 0) {
+                endpoint = $"{Options.UserInformationEndpoint}/~:({string.Join(",", Options.Fields)})";
```

Comment:
Manual concatenation is still a bit fragile when manipulating user-defined URLs. Please consider using `QueryHelpers` instead (e.g https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/03599ad33a0bd4bfd3b0ef42f4b0670ae6f21680/src/AspNet.Security.OAuth.Instagram/InstagramAuthenticationHandler.cs#L33)